auth/store.go

```diff
@@ -887,7 +969,7 @@ func (as *authStore) isAuthEnabled() bool {
 	return as.enabled
 }
 
-func NewAuthStore(be backend.Backend, indexWaiter func(uint64) <-chan struct{}) *authStore {
+func NewAuthStore(be backend.Backend, indexWaiter func(uint64) <-chan struct{}, tokenType, jwtSignMethod, jwtPubKeyPath, jwtPrivKeyPath string) *authStore {
```

Comment:
Might make sense to have the cut here by having a `TokenProvider` instead of the token type stuff:
```go
type TokenProvider interface {
    Info(token string) (*AuthInfo, bool) // replaces AuthInfoFromToken
    Assign(ctx context.Context, username string) (string, error) // called by Authenticate; returns token
}
```
implemented by:
```go
type TokenJWT struct { ... }
type TokenSimple struct  {...}
```
which will have constructors `NewTokenJWT` / `NewTokenSimple` instead of building them in `NewAuthStore()`